Build Error with MPI and C++ Examples Enabled

The build command `cmake -DENABLE_MPI=ON -DEXAMPLES_ENABLE_CXX=ON ..` causes the following build error for me
```
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o: In function `MPI::Op::Init(void (*)(void const*, void*, int, MPI::Datatype const&), bool)':                                        
/usr/include/openmpi-x86_64/openmpi/ompi/mpi/cxx/op_inln.h:121: undefined reference to `ompi_mpi_cxx_op_intercept'                                                                         
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o: In function `MPI::Intracomm::Clone() const':                                                                                       
/usr/include/openmpi-x86_64/openmpi/ompi/mpi/cxx/intracomm_inln.h:23: undefined reference to `MPI::Comm::Comm()'                                                                           
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o: In function `MPI::Graphcomm::Clone() const':                                                                                       
/usr/include/openmpi-x86_64/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'                                                                                
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o: In function `MPI::Cartcomm::Sub(bool const*) const':                                                                               
/usr/include/openmpi-x86_64/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'                                                                                
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o: In function `MPI::Intracomm::Create_graph(int, int const*, int const*, bool) const':                                               
/usr/include/openmpi-x86_64/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'                                                                                
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o: In function `MPI::Cartcomm::Clone() const':                                                                                        
/usr/include/openmpi-x86_64/openmpi/ompi/mpi/cxx/intracomm.h:25: undefined reference to `MPI::Comm::Comm()'                                                                                
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o:/usr/include/openmpi-x86_64/openmpi/ompi/mpi/cxx/intracomm_inln.h:23: more undefined references to `MPI::Comm::Comm()' follow       
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o:(.rodata._ZTVN3MPI8DatatypeE[_ZTVN3MPI8DatatypeE]+0x78): undefined reference to `MPI::Datatype::Free()'                             
CMakeFiles/ark_analytic_sys.dir/ark_analytic_sys.cpp.o:(.rodata._ZTVN3MPI3WinE[_ZTVN3MPI3WinE]+0x48): undefined reference to `MPI::Win::Free()'                                            
collect2: error: ld returned 1 exit status
make[2]: *** [examples/arkode/CXX_serial/CMakeFiles/ark_analytic_sys.dir/build.make:102: examples/arkode/CXX_serial/ark_analytic_sys] Error 1                                              
make[1]: *** [CMakeFiles/Makefile2:6810: examples/arkode/CXX_serial/CMakeFiles/ark_analytic_sys.dir/all] Error 2                                                                           
make: *** [Makefile:146: all] Error 2
```

I'm using gcc 8.5 and Open MPI 4.1.1. With mpich 4.1.1 it compiles without error.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -30,6 +30,8 @@ Fixed a bug that caused error messages to be cut off in some cases. Fixes [GitHu
 
 Fixed a memory leak when an error handler was added to a `SUNContext`. Fixes [GitHub Issue #466](https://github.com/LLNL/sundials/issues/466).
 
+Fixed a CMake bug that caused an MPI linking error for our C++ examples in some instances. Fixes [GitHub Issue #464](https://github.com/LLNL/sundials/issues/464).
+
 ## Changes to SUNDIALS in release v7.0.0
 
 ### Major Feature

--- a/doc/shared/RecentChanges.rst
+++ b/doc/shared/RecentChanges.rst
@@ -29,3 +29,4 @@ Fixed a bug that caused error messages to be cut off in some cases. Fixes `GitHu
 
 Fixed a memory leak when an error handler was added to a :c:type:`SUNContext`. Fixes `GitHub Issue #466 <https://github.com/LLNL/sundials/issues/466>`_.
 
+Fixed a CMake bug that caused an MPI linking error for our C++ examples in some instances. Fixes `GitHub Issue #464 <https://github.com/LLNL/sundials/issues/464>`_.

--- a/src/sundials/CMakeLists.txt
+++ b/src/sundials/CMakeLists.txt
@@ -114,8 +114,11 @@ endif()
 add_prefix(${SUNDIALS_SOURCE_DIR}/src/sundials/ sundials_SOURCES)
 
 if(ENABLE_MPI)
-  set(_link_mpi_if_needed PUBLIC MPI::MPI_C)
+  set(_link_mpi_if_needed PUBLIC 
+    MPI::MPI_C
+    $<$<LINK_LANGUAGE:CXX>:MPI::MPI_CXX>)
 endif()
+
 
 if(SUNDIALS_BUILD_WITH_PROFILING)
   if(ENABLE_CALIPER)

